scopes/ui-foundation/ui/ui.main.runtime.ts

```diff
@@ -446,10 +453,15 @@ export class UiMain {
       config || this.harmony.config.toObject(),
       ignoreVersion
     );
-    const filepath = resolve(join(path || __dirname, `${runtimeName}.root${sha1(contents)}.js`));
-    if (fs.existsSync(filepath)) return filepath;
+    const filename = `${runtimeName}.root${sha1(contents)}.js`;
+    // Write the file into `./node_modules/.cache/bit/teambit.ui/` if the path
+    // is provided, which is usually the workspace root.
+    const relativeFilepath = path ? `./node_modules/.cache/bit/teambit.ui/${filename}` : filename;
+    const filepath = resolve(join(path || __dirname, relativeFilepath));
+    const returnedPath = path ? relativeFilepath : filepath;
```

Comment:
This is a breaking API for the `generateRoot` API and i'm pretty sure will break bit cloud which uses it.
before it it used the path as is, now you add another internal path into it. maybe you should instead doing that outside the func.

Reply:
@GiladShoham understood. I moved the addition of `./node_modules/.cache/bit/teambit.ui/` into `PreviewMain -> writePreviewRuntime()` and reverted other `generateRoot()` calls. Please take a look further. Thanks.